packages/http-server-fastify/src/FastifyHttpServer.ts

```diff
@@ -61,16 +61,24 @@ export class FastifyHttpServer extends HttpServerModule<{
 		this.initHttpServer();
 		await this.registerMiddlewares();
 		await this.registerPlugins();
-
 		await super.createRoutes();
 	}
 
 	async onInit() {
+		if (!this.instance) {
```

Comment:
Do we nullify them? I think the problem is precisely that the instance is closed but keeps the reference

Reply:
Ahh right sorry. So the thing is we need to do the instantiation process onRegister (this is needed for the openapiModule). Then for the case where we shutdown the apl and re-init, we want this process to happen on the init (since the onRegister won't be called). So the if prevents that the first time the instance is created twice (onRegister and onIntit). That is my understanding.

Reply:
right I see now.
To me it seems something that probably should be treated as first-class citizen on the app itself, as other modules may require it.

What about a flag on the App init method that force triggering the onRegister hooks for the modules? It would be ideal for testing
```
app.init({ runOnRegisterHooks: true })
```

Reply:
My first idea was that the initialization of fastify should happen onInit, not onRegister. But I couldn't make it work in combination with openapi. Idk how you like the idea, but I'm not sure it's possible. 

Otherwise this seems like a way simpler solution.

Reply:
The heavy lifting of the modules should be performed during the registration phase.
The init phase should just start connections and servers.

This is needed as you mentioned, other modules may attach behaviours to other modules, like OpenAPI.

Reply:
a simpler logic, inside the app.init() method could be:

if the app status is `destroyed`, rerun the onRegister hooks for all modules

Reply:
Went for the second idea